src/utils/test-runner.js

```diff
@@ -47,33 +28,61 @@ function validSourceApiVersion() {
     }
 }
 
+function getJestPath() {
+    const packageJsonPath = require.resolve('jest/package.json');
+
+    const { bin } = require(packageJsonPath);
+    return path.resolve(path.dirname(packageJsonPath), bin);
+}
+
+function getJestArgs(argv) {
+    // Extract known options from parsed arguments
+    const knownOptions = Object.keys(argv)
+        .filter((key) => argv[key] && JEST_PASSTHROUGH_OPTIONS.has(key))
+        .map((key) => `--${key}`);
+
+    // Extract remaining options after `--`
+    const rest = argv._;
+
+    const jestArgs = [...knownOptions, ...rest];
+
+    // Force jest to run in band when debugging.
+    if (argv.debug) {
+        jestArgs.unshift('--runInBand');
+    }
+
+    // Provide default configuration when none is present at the project root.
+    const hasCustomConfig = fs.existsSync(path.resolve(PROJECT_ROOT, 'jest.config.js'));
+    if (!hasCustomConfig) {
+        jestArgs.unshift(`--config=${JSON.stringify(jestConfig)}`);
+    }
+
+    return jestArgs;
+}
+
 async function testRunner(argv) {
     if (!argv.skipApiVersionCheck) {
         validSourceApiVersion();
     }
 
-    const hasCustomConfig = fs.existsSync(path.resolve(PROJECT_ROOT, 'jest.config.js'));
-    const config = hasCustomConfig ? [] : ['--config', JSON.stringify(jestConfig)];
+    const spawnCommand = 'node';
+    const spawnArgs = [getJestPath(), ...getJestArgs(argv)];
 
-    const options = getOptions(argv);
     if (argv.debug) {
+        spawnArgs.unshift('--inspect-brk');
+
         info('Running in debug mode...');
-        let commandArgs = ['--inspect-brk', jestPath, '--runInBand'];
-        commandArgs = commandArgs.concat(options);
-        if (!hasCustomConfig) {
-            commandArgs.push('--config=' + JSON.stringify(jestConfig));
-        }
-        const command = 'node';
-        info(command + ' ' + commandArgs.join(' '));
-
-        shell.runCommand(command, commandArgs);
-    } else {
-        // Jest will not set the env if not run from the bin executable
-        if (process.env.NODE_ENV == null) {
-            process.env.NODE_ENV = 'test';
-        }
-        jestRunner.run([...config, ...options]);
+        info(`${spawnCommand} ${spawnArgs.join(' ')}`);
```

Comment:
This logging seems useful outside of debug mode, should we move it outside of this if-block?

Reply:
I would prefer to keep this in debug mode only. This avoids cluttering the console when you know that jest is properly setup.